CIScripts/Testenv/Testenv.Impl.ps1

```diff
@@ -4,18 +4,64 @@ class Testenv {
     [ControllerConfig] $Controller
     [Testbed[]] $Testbeds
 
-    [PSSessionT[]] $Sessions
+    [PSSessionT[]] $Sessions = $null
 
-    [Void] Initialize() {
-        $this.Sessions = New-RemoteSessions -VMs $this.Testbeds
-        Set-ConfAndLogDir -Sessions $this.Sessions
-        Sync-MicrosoftDockerImagesOnTestbeds -Sessions $this.Sessions
-    }
+    [System.Collections.Stack] $CleanupStacks = [System.Collections.Stack]::new()
+    [MultiNode] $Multinode = $null
+    [LogSource[]] $LogSources = $null
+    [ContrailRepo] $ContrailRepo = $null
 
     Testenv([String] $TestenvConfFile) {
         $this.System = [SystemConfig]::LoadFromFile($TestenvConfFile)
         $this.OpenStack = [OpenStackConfig]::LoadFromFile($TestenvConfFile)
         $this.Controller = [ControllerConfig]::LoadFromFile($TestenvConfFile)
         $this.Testbeds = [Testbed]::LoadFromFile($TestenvConfFile)
     }
+
+    [Void] Initialize() {
+        $CleanupStack = $this.NewCleanupStack()
+        $this.Sessions = New-RemoteSessions -VMs $this.Testbeds
+        $CleanupStack.Push({Param([PSSessionT[]] $Sessions) Remove-PSSession $Sessions}, @(, $this.Sessions))
+        Set-ConfAndLogDir -Sessions $this.Sessions
+        Sync-MicrosoftDockerImagesOnTestbeds -Sessions $this.Sessions
+    }
+
+    [Void] Initialize_New([String] $LogDir, [String] $ContrailProject, [Bool] $InstallComponents) {
```

Comment:
This method will be joined with Initialize after all tests will be refactored

Reply:
Add TODO comment please